app/jobs/search_keyword_job.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class SearchKeywordJob < ApplicationJob
+  def perform(keyword_id)
+    keyword = Keyword.includes(:source).where(keywords: { id: keyword_id }).first
+    keyword_name = keyword.name
+    source_name = keyword.source.name.downcase
+    case source_name
+    when 'google'
+      search_result = Google::SearchService.new(keyword_name).call
+    when 'bing'
+      search_result = Bing::SearchService.new(keyword_name).call
+    end
```

Comment:
```suggestion
    search_result = search_service(keyword).new(keyword_name).call
```

```
private

def search_service(keyword)
  source_name = keyword.source.name.downcase

  return Bing::SearchService if source_name == 'bing'

  Google::SearchService
end
```

What about we extract the concern that retrieves the service class into a dedicated method?
As you might have seen Ruby loves "small methods" :D

Reply:
That's a good case for Factory

Reply:
solved in c69e184